python/whylogs/api/logger/result_set.py

```diff
@@ -90,6 +91,11 @@ def profile(self) -> Optional[DatasetProfile]:
     def get_writables(self) -> Optional[List[Writable]]:
         return [self.view()]
 
+    def set_dataset_timestamp(self, dataset_timestamp: datetime) -> None:
+        profile = self.profile()
+        if profile:
```

Comment:
profile is not None or is not empty?

Reply:
Hmm thinking about this more, because this is in the base class, it would be less sneaky to allow the exception here and not check?